### PR TITLE
Update packet

### PR DESCRIPTION
**Pull request by sctnightcore**
https://github.com/HerculesWS/Hercules/pull/2176
ty @4144 ♥ / Herc Dev 

```
* New Packet 'login_pin_code_request'
* New Packet 'show_eq'
```

### CODE REVIEW

**Comment by Airdrop9999** [comment]
hi 
gg
 i have change all the pm files and still this is showing

File: src/Network/Receive.pm [hunk 1]
```diff
@@ -2331,7 +2331,7 @@ sub show_eq {
 			keys => [qw(ID nameID type type_equip equipped upgrade cards expire bindOnEquipType sprite_id identified)],
 		};
 		
-	} elsif ($args->{switch} eq '0A2D') { # PACKETVER >= 20150226
+	} elsif ($args->{switch} eq '0A2D' && $args->{switch} eq '0B03') { # PACKETVER >= 20150226 / PACKETVER_RE_NUM >= 20180801
```
Comment (alisonrag) [inline src/Network/Receive.pm]: should be || (or)
&& is equal to AND

no one packet have 2 IDS

```
 elsif ($args->{switch} eq '0A2D' || $args->{switch} eq '0B03')
```

File: src/Network/Receive/ServerType0.pm [hunk 2]
```diff
@@ -4565,5 +4570,10 @@ sub progress_bar_unit {
 	debug "Displays progress bar (GID: $args-{GID} time: $args-{time})\n";	
 }
 
+#TODO
+#0AFD
+sub guild_member_position {
```
Comment (alisonrag) [inline src/Network/Receive/ServerType0.pm]: add this to receive.pm to avoid duplicated code, than i can approve

Comment (alisonrag) [inline src/Network/Receive/ServerType0.pm]: add this to receive.pm to avoid duplicated code, than i can approve

File: src/Network/Receive.pm [hunk 3]
```diff
@@ -4434,16 +4434,20 @@ sub guild_location {
 
 sub guild_leave {
 	my ($self, $args) = @_;
-
+	my $name = bytesToString($args->{name}) if defined $args->{name};
+	my $gid = unpack("V",$args->{ID}) if defined $args->{ID};
+	
 	message TF("%s has left the guild.\n" .
-		"Reason: %s\n", bytesToString($args->{name}), bytesToString($args->{message})), "schat";
+		"Reason: %s\n", $name ? $gid, bytesToString($args->{message})), "schat";
```
Comment (Nipodemos) [inline src/Network/Receive.pm]: this syntax here is wrong...
should be `$name ? $gid : ''`

**Pull request closed without merging**